src/libraries/System.Linq.Parallel/src/System/Linq/Parallel/Channels/AsynchronousChannel.cs

```diff
@@ -198,6 +199,7 @@ internal bool IsDone
         // Used by a producer to flush out any internal buffers that have been accumulating
         // data, but which hasn't yet been published to the consumer.
 
+        [UnsupportedOSPlatform("browser")]
```

Comment:
System.Linq.Parallel _does_ work today for the most part though, it just doesn't run in parallel since it uses `Environment.ProcessorCount` degree of parallelism by default which is 1 for wasm (same as if you had e.g. a single-core PC). I'm not sure marking the whole assembly as unsupported on browser is the way to go.

Reply:
There's a ton of code that can't be trimmed and ends up being a nop, plus a chance that you will hit an unsupported operation.  If we believe the library is useful and should be available in a browser context, I think it'd be better to have a simple, dedicated implementation that is a pure passthrough to Enumerable and that's used for browser.

Reply:
I agree with that but instead of doing it here, we should just add the attribute to implementation assembly and remove it in PR which forwards to System.Linq for wasm.

Reply:
I'm fine with that.

Reply:
Marked entire assembly unsupported

Reply:
Do we have a tracking issue for replacing System.Linq.Parallel implementation with one that just forwards to Enumerable?

Reply:
We have this one https://github.com/dotnet/runtime/issues/43752